src/codegen/mod.rs

```diff
@@ -265,23 +277,31 @@ impl CodeGenerator for Var {
                ctx: &BindgenContext,
                result: &mut CodegenResult,
                item: &Item) {
-        let name = item.canonical_name(ctx);
+        let canonical_name = item.canonical_name(ctx);
+
+        // TODO: Maybe warn here if there's a type mismatch
```

Comment:
Yeah, from what I remember bindgen prints clang warnings, so there is no need to do an other warning